mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.tmpl

```diff
@@ -13278,3 +13355,63 @@ resource "google_container_cluster" "with_enterprise_config" {
 }
 `, projectID, clusterName, networkName, subnetworkName)
 }
+
+{{ if ne $.TargetVersionName `ga` -}}
+func testAccContainerCluster_withPodAutoscalingConfig(projectID, name, networkName, subnetworkName string, enable bool) string {
+
+  podAutoscalingBlock := ""
+  if enable { 
+    podAutoscalingBlock=`
+    pod_autoscaling {
+      hpa_profile = "NONE"
+    }
+  `
+  }
+
+  return fmt.Sprintf(`
+data "google_project" "project" {
+  provider   = google-beta
+  project_id = "%s"
+}
+resource "google_compute_network" "network" {
+  provider                 = google-beta
+  name                     = "%s"
+  auto_create_subnetworks  = false
+}
+resource "google_compute_subnetwork" "subnet" {
+  provider      = google-beta
+  name          = "%s"
+  network       = google_compute_network.network.id
+  ip_cidr_range = "10.12.22.0/24"
+  region        = "us-central1"
+}
+resource "google_container_cluster" "with_pod_autoscaling_config" {
+  provider           = google-beta
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  network            = google_compute_network.network.name
+  subnetwork         = google_compute_subnetwork.subnet.name
+  deletion_protection = false
+  %v
+}
+`, projectID, networkName, subnetworkName, name, podAutoscalingBlock)
+}
+
+func testAccContainerCluster_withPodAutoscalingConfigAutopilot(projectID, name string) string {
+  return fmt.Sprintf(`
+  data "google_project" "project" {
+    provider = google-beta
+    project_id = "%s"
+  }
+  resource "google_container_cluster" "with_pod_autoscaling_config_autopilot" {
+    provider = google-beta
+    name               = "%s"
+    location           = "us-central1"
+    enable_autopilot = true
+    min_master_version = "1.32.1-gke.1729000"
```

Comment:
use the [google_container_engine_versions](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/container_engine_versions) datasource instead of hardcoding a version; otherwise this test will break in the future when this version is removed.